compiler/rustc_parse/src/parser/pat.rs

```diff
@@ -782,6 +788,18 @@ impl<'a> Parser<'a> {
             .emit();
     }
 
+    fn error_inclusive_range_match_arrow(&self, span: Span) {
+        let without_eq = span.with_hi(span.hi() - rustc_span::BytePos(1));
+        self.struct_span_err(span, "unexpected `=>` after open range")
+            .span_suggestion_verbose(
+                without_eq,
+                "add a space between the pattern and `=>`",
+                ".. ",
```

Comment:
We could probably adjust the spans a bit more carefully to just make the suggestion `" "`. Do you mind doing that?